wandb/sdk/wandb_run.py

```diff
@@ -355,10 +355,12 @@ def _init(
         config: Optional[Dict[str, Any]] = None,
         sweep_config: Optional[Dict[str, Any]] = None,
     ) -> None:
+
+        self._settings = settings
         self._config = wandb_config.Config()
         self._config._set_callback(self._config_callback)
         self._config._set_artifact_callback(self._config_artifact_callback)
-        self._config._set_settings(settings)
+        self._config._set_settings(self._settings)
```

Comment:
Why weren't we doing this before?

Reply:
hmm... i don't think it matters since we assigned `settings` to `self._settings` later on... i just moved the assignment to the top of the function, this is should be the same.